src/ImageSharp/Common/Memory/BufferPointer.cs

```diff
@@ -5,25 +5,19 @@
 
 namespace ImageSharp
 {
+    using System.Numerics;
     using System.Runtime.CompilerServices;
+    using System.Runtime.InteropServices;
 
     /// <summary>
     /// Utility methods for <see cref="BufferPointer{T}"/>
     /// </summary>
     internal static class BufferPointer
     {
         /// <summary>
-        /// Gets a <see cref="BufferPointer{T}"/> to the beginning of the raw data in 'buffer'.
+        /// It's worth to use Marshal.Copy() over this size.
         /// </summary>
-        /// <typeparam name="T">The element type</typeparam>
-        /// <param name="buffer">The input <see cref="PinnedBuffer{T}"/></param>
-        /// <returns>The <see cref="BufferPointer{T}"/></returns>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe BufferPointer<T> Slice<T>(this PinnedBuffer<T> buffer)
-            where T : struct
-        {
-            return new BufferPointer<T>(buffer.Array, (void*)buffer.Pointer);
-        }
+        private const uint ByteCountThreshold = 1024u;
```

Comment:
Why is it better to use `Marshal.Copy()` over that size? why wouldn't you use it less that that?

How was this number decided?